packages/performance/src/test/performance.test.ts

```diff
@@ -12,4 +16,47 @@ describe('Performance', () => {
     performance.finish();
     expect(spy).toHaveBeenCalledWith(expect.any(Navigation));
   });
+
+  describe.skip('first-input-delay', () => {
+    class MockPerfMetrics implements PerfMetrics {
+      private _callback: FirstInputDelayCallback | null = null;
+
+      get callback() {
+        return this._callback;
+      }
+
+      onFirstInputDelay(callback) {
+        this._callback = callback;
+      }
+    }
+
+    it('registers a first-input-delay listener', () => {
+      const perfMetrics = new MockPerfMetrics();
+      window.perfMetrics = perfMetrics;
+
+      // eslint-disable-next-line no-new
+      new Performance();
+      expect(perfMetrics.callback).not.toBeNull();
+    });
+
+    it('sends first-interaction-delay lifecycle event', () => {
+      const perfMetrics = new MockPerfMetrics();
+      window.perfMetrics = perfMetrics;
+
+      const lifecycleSpy = jest.fn();
+      // eslint-disable-next-line no-new
+      const performance = new Performance();
+      performance.on('lifecycleEvent', lifecycleSpy);
+
+      // eslint-disable-next-line typescript/no-non-null-assertion
+      perfMetrics.callback!(1000, new Event('click'));
```

Comment:
Oh interesting, I wonder if we should capture the event type as metadata for the `fid` event?

Reply:
Let's get some results in first (aka, _I'd like to ship this now_ 😛 )